lib/native/source/opensmt/api.patch

```diff
@@ -0,0 +1,537 @@
+         char* tmp_str;
+         opensmt::stringToRational(tmp_str, sym_store.getName(getPterm(tr).symb()));
+         opensmt::Number v(tmp_str);
+-        if (!isNonNegNumConst(tr)) {
++        if (!isNonNegative(v)) {
+             v.negate();
+             is_neg = true;
+         }
+diff --git a/src/logics/ArithLogic.h b/src/logics/ArithLogic.h
+index 6f2669f9..fa8a7e26 100644
+--- a/src/logics/ArithLogic.h
++++ b/src/logics/ArithLogic.h
+@@ -4,7 +4,7 @@
+ #include "Number.h"
+ #include <numeric>
+ 
+-class LANonLinearException : std::runtime_error
++class LANonLinearException : public std::runtime_error
+ {
+     std::string msg;
+ public:
+@@ -417,4 +417,4 @@ public:
+     bool operator ()  (PTRef x_, PTRef y_) const;
+ 
+ };
+-#endif
+\ No newline at end of file
++#endif
+diff --git a/src/logics/FunctionTools.h b/src/logics/FunctionTools.h
+index fc9dbfb7..943a466d 100644
+--- a/src/logics/FunctionTools.h
++++ b/src/logics/FunctionTools.h
+@@ -33,7 +33,7 @@ class TemplateFunction {
+     FunctionSignature signature;
+     PTRef tr_body;
+ 
+-    inline static std::string_view constexpr template_arg_prefix = ".arg";
++    inline static constexpr std::string_view template_arg_prefix = ".arg";
+     inline static std::size_t template_arg_counter = 0;
+ public:
+     static std::string nextFreeArgumentName() { return std::string(template_arg_prefix) + std::to_string(template_arg_counter++); }
+diff --git a/src/logics/Logic.cc b/src/logics/Logic.cc
+index b90430c2..1161b068 100644
+--- a/src/logics/Logic.cc
++++ b/src/logics/Logic.cc
+@@ -15,6 +15,7 @@
+ #include "Substitutor.h"
+ #include "smt2tokens.h"
+ #include "IteHandler.h"
++#include "Interpret.h"
+ 
+ #include <iostream>
+ #include <map>
+@@ -32,7 +33,7 @@ const char* Logic::e_argnum_mismatch = "incorrect number of arguments";
+ const char* Logic::e_bad_constant    = "incorrect constant for logic";
+ 
+ const char* Logic::tk_val_uf_default   = "UFDefault";
+-const char* Logic::tk_val_bool_default = "true";
++const char* Logic::tk_val_bool_default = "false";
+ 
+ const char* Logic::tk_true     = "true";
+ const char* Logic::tk_false    = "false";
+@@ -130,12 +131,13 @@ bool Logic::hasQuotableChars(std::string const & name) const
+         0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+         0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+         0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
++        0, 0, 0, 0, 0, 0
+     };
+ 
+     if (name.front() == '|' and name.back() == '|') return false; // Already quoted
+ 
+-    for (auto c : name) {
+-        if (quotable[(int)c])
++    for (unsigned char c : name) {
++        if (quotable[c])
+             return true;
+     }
+     return false;
+@@ -150,7 +152,7 @@ std::string Logic::disambiguateName(std::string const & protectedName, SRef sort
+         return protectedName;
+     }
+ 
+-    auto isQuoted = [](std::string const &s) { return s.size() > 3 and *s.begin() == '|' and *(s.end()-1) == '|'; };
```

Comment:
Why do we need to patch the number of chars for quoting?

Reply:
I think this is another bug, but I'll have to check with Martin.

Reply:
Is the terminal \0 counted in the string-size in C++? Then 3 would be valid for "||\0".

Reply:
No, it is not counted in the size of `std::string`. I'll see if I can find out why it is like this, but I agree it should probably be `2`.